brigade/plugins/tasks/networking/netmiko_tasks.py

```diff
@@ -0,0 +1,161 @@
+        objs.append(temp_dict)
+    return objs
+
+
+def netmiko_args(task, ip=None, host=None, username=None, password=None, device_type=None,
+                 netmiko_dict=None):
+    """Process arguments coming from different sources."""
+    parameters = {
+        "username": username or task.host["brigade_username"],
+        "password": password or task.host["brigade_password"],
+    }
+
+    if host is None and ip is None:
+        parameters["ip"] = task.host["brigade_ip"]
+    elif ip is not None:
+        parameters["ip"] = ip
+    elif host is not None:
+        parameters["host"] = host
+
+    if netmiko_dict is not None:
+        parameters.update(netmiko_dict)
+    else:
+        netmiko_dict = {}
+
+    if device_type is None:
+        if netmiko_dict.get("device_type") is None:
+            device_type = task.host["nos"]
+        else:
+            device_type = netmiko_dict.get("device_type")
+
+    # Convert to netmiko device_type format (if napalm format is used)
+    parameters['device_type'] = napalm_to_netmiko_map.get(device_type, device_type)
+    return parameters
+
+
+def netmiko_run(task, method, ip=None, host=None, username=None, password=None,
+                device_type=None, netmiko_dict=None, cmd_args=None, cmd_kwargs=None):
+    """
+    Execute any Netmiko method from connection class (BaseConnection class and children).
+
+    Arguments:
+        method(str): Netmiko method to use
+        ip (string, optional): defaults to ``brigade_ip``
+        host (string, optional): defaults to None
+        username (string, optional): defaults to ``brigade_username``
+        password (string, optional): defaults to ``brigade_password``
+        device_type (string, optional): Netmiko device_type to use, defaults to ``nos`` (mapped \
+        through napalm_to_netmiko_map)
+        netmiko_dict (dict, optional): Additional arguments to pass to Netmiko ConnectHandler, \
+        defaults to None
+
+    Returns:
+        :obj:`brigade.core.task.Result`:
+          * result (``dict``): dictionary with the result of the getter
+    """
+    parameters = netmiko_args(task=task, ip=ip, host=host, username=username,
+                              password=password, device_type=device_type,
+                              netmiko_dict=netmiko_dict)
+    with ConnectHandler(**parameters) as net_connect:
+        netmiko_method = getattr(net_connect, method)
+        if cmd_args is None:
+            cmd_args = ()
+        if cmd_kwargs is None:
+            cmd_kwargs = {}
+        result = netmiko_method(*cmd_args, **cmd_kwargs)
+    return Result(host=task.host, result=result)
+
+
+def netmiko_send_command(task, ip=None, host=None, username=None, password=None,
+                         device_type=None, netmiko_dict=None, use_textfsm=False,
+                         cmd_args=None, cmd_kwargs=None):
+    parameters = netmiko_args(task=task, ip=ip, host=host, username=username,
```

Comment:
If this is a second task, please, use different files for different tasks. Mostly because I haven't really come up with a proper solution for `connection: remote` tasks and for those it might be convenient to just scp the file and execute remotely.

Reply:
Why? I think brigade shouldn't care...i.e. don't impose artificial constraints at the Python level. 

I probably don't care about remote tasks (and don't want to recreate Ansible and its associated remote task problems)...so I am viewing these as entirely local tasks.

Reply:
Yeah, I agree with you but some tasks might make sense to run them remotely. For instance, a task to manage `rpms/deb` packages. So basically enforcing one task per file would mostly be for consistency. It also helps identifying which tasks are untested by checking the coverage report but this is a minor thing.

But as you hinted already, this is an artificial constraint, doing it or not doesn't really change anything and we might still change our minds a few times in the next few weeks so what I have been doing is adding the tasks here:

https://github.com/napalm-automation/brigade/blob/develop/brigade/plugins/tasks/networking/__init__.py

That way we can keep changing our mind as users should be getting the tasks as:

```
from brigade.plugins.tasks import networking

networking.netmiko_run
```

In summary, it doesn't really matter, keep them in the same file for now if you want. Just make sure you add the tasks to the `__init__` file so they can be imported at the package level in case we decide to move them around at some point.